File: Kotlin/src/main/kotlin/c_compound_conditions/exercise_32.kt
package c_compound_conditions

import kotlin.random.Random

// [CHALLENGE] Create a game where the computer will draw a number between 1 and 5, and the player will try to guess the drawn value.

fun main() {
    val draw = Random.nextInt(1, 6)

    print("Type a number between 1 and 5: ")

    when (val number = readln().toInt()) {
        draw -> {
            println("Number: $number; drawn value: $draw.")
            println("You hit the drawn value.")
        }
        in 1..5 -> {
            println("Number: $number; drawn value: $draw.")
            println("Try again.")
        }
        else -> {
            println("Invalid.")
        }
    }
}

File: Dart/c_compound_conditions/exercise_32.dart
// [CHALLENGE] Create a game where the computer will draw a number between 1 and 5, and the player will try to guess the drawn value.

import 'dart:io';
import 'dart:math';

void main() {
  String answer = 'yes';

  while (answer == 'yes') {
    int computerDraw = Random().nextInt(5) + 1;
    int playerChoice = readInt('Type a number between 1 and 5: ');

    printMessage(playerChoice, computerDraw);

    stdout.write('\ndo you want to try again? ');
    answer = stdin.readLineSync()!.toLowerCase();
  }
}

int readInt(String prompt) {
  int? value;

  do {
    stdout.write(prompt);
    String? input = stdin.readLineSync();
    value = int.tryParse(input!);

    if (value == null || value < 1 || value > 5) {
      print('Error: Please enter a valid number.');
    }
  } while (value == null || value < 1 || value > 5);

  return value;
}

void printMessage(int playerChoice, int computerDraw) {
  if (computerDraw == playerChoice) {
    print(
        'YOU GUESSED IT!\nNumber: $playerChoice — Drawn value: $computerDraw.');
  } else {
    print('WRONG GUESS!\nNumber: $playerChoice — Drawn value: $computerDraw.');
  }
}
